datafusion/expr/src/logical_plan/builder.rs

```diff
@@ -1507,6 +1505,7 @@ fn table_source(table_schema: &Schema) -> Arc<dyn TableSource> {
 pub fn wrap_projection_for_join_if_necessary(
     join_keys: &[Expr],
     input: LogicalPlan,
+    upper_exprs: &[&Expr],
```

Comment:
Could you please add some documentation about what this parameter is being used for? 

For example, it seems like it represents the expressions required by the parent join. And the change is that this function ensures that all expressions in `upper_exprs` are included in the output `LogicalPlan`?

Reply:
sure, I'll add that